articles/tokens/concepts/token-storage.md

```diff
@@ -0,0 +1,104 @@
+
+## Traditional web app scenarios
+
+If your app is using a sign in scenario that doesn't require API calls, only an ID Token is required. There is no need to store it. You can validate it and get the data from it that you required. 
+
+If your app needs to call APIs on behalf of the user, Access Tokens and (optionally) Refresh Tokens are needed. These can be stored server-side or in a session cookie. The cookie needs to be encrypted and have a maximum size of 4 KB. If the data to be stored is large, storing tokens in the session cookie is not a viable option. 
+
+Use the following flow types in these scenarios: 
+
+- [Authorization Code Flow](/flows/concepts/auth-code)
+- [Regular Web App Quickstarts](/quickstart/webapp)
+
+## Native/Mobile app scenarios
+
+Store tokens in a secure storage that the OS offers and limit access to that storage. For example, leverage KeyStore for Android and KeyChain for iOS.
+
+Use the following flow types in these scenarios:
+
+- [Authorization Code Flow with Proof Key for Code Exchange](/flows/concepts/auth-code-pkce)
+- [Saving and Renewing Tokens for Android](/libraries/auth0-android/save-and-refresh-tokens)
+- [Saving and Renewing Tokens for Swift](/libraries/auth0-swift/save-and-refresh-jwt-tokens)
+- [Native/Mobile Apps Quickstarts](/quickstart/native)
+
+## Single-page app scenarios
+
+We recommend using the [Auth0 Single Page App SDK](/libraries/auth0-spa-js). The Auth0 SPA SDK handles token storage, session management, and other details for you.
+
+When the SPA calls only an API that is served from a domain that can share cookies with the domain of the SPA, no tokens are needed. OAuth adds additional attack vectors without providing any additional value and should be avoided in favor of a traditional cookie-based approach. For an overview of this approach and an example implementation, see [Single-Page App Authentication Using Cookies](/login/spa/authenticate-with-cookies).
+
+When the SPA calls multiple APIs that reside in a different domain, access and optionally refresh tokens are needed.
+
+-  If the SPA backend can handle the API calls, handle tokens server-side using:
+    - [Authorization Code Flow](/flows/concepts/auth-code)
+    - [Authorization Code Flow with Proof Key for Code Exchange (PKCE)](/flows/concepts/auth-code-pkce)
+    - [Hybrid Flow](/api-auth/grant/hybrid)
+
+- If the SPA backend cannot handle the API calls, the tokens should be stored in the SPA backend but the SPA needs to fetch the tokens from the backend to perform requests to the API. A protocol needs to be established between the backend and the SPA to allow the secure transfer of the token from the backend to the SPA.
+
+- If you have a SPA with **no** corresponding backend server, your SPA should request new tokens on login and store them in memory without any persistence. To make API calls, your SPA would then use the in-memory copy of the token.
```

Comment:
> If you have a SPA

Throughout the docs there's a mix of "a SPA" and "an SPA" - I recommend having a sweep through and picking one style or the other, for consistency.

Reply:
Good catch. I'll do that.